src/Symfony/Component/Serializer/NameConverter/CamelCaseToSnakeCaseNameConverter.php

```diff
@@ -48,7 +48,7 @@ public function normalize($propertyName)
 
             $len = strlen($propertyName);
             for ($i = 0; $i < $len; ++$i) {
-                if (ctype_upper($propertyName[$i])) {
+                if (ctype_upper($propertyName[$i]) && $i !== 0) {
```

Comment:
I ran a small php benchmarks with 100 000 occurrences and different names:
with ``Foo``:
```
test_original             : 0.099 sec.
test_energetick           : 0.047 sec.
test_dunglas              : 0.077 sec.
```
with ``aLongNameIsVeryLong``:
```
test_original             : 0.558 sec.
test_energetick           : 0.364 sec.
test_dunglas              : 0.480 sec.
```

The results aren't that significant so choose what you think is the more readable :-)